index.test-d.ts

```diff
@@ -3,33 +3,57 @@ import camelcaseKeys = require('.');
 
 const fooBarObject = {'foo-bar': true};
 const camelFooBarObject = camelcaseKeys(fooBarObject);
-expectType<typeof fooBarObject>(camelFooBarObject);
+expectType<{ fooBar: boolean }>(camelFooBarObject);
 
 const fooBarArray = [{'foo-bar': true}];
 const camelFooBarArray = camelcaseKeys(fooBarArray);
-expectType<typeof fooBarArray>(camelFooBarArray);
+expectType<Array<{ fooBar: boolean }>>(camelFooBarArray);
 
-expectType<Array<{[key in 'foo-bar']: true}>>(camelcaseKeys([{'foo-bar': true}]));
+expectType<Array<{[key in 'fooBar']: boolean}>>(camelcaseKeys([{'foo-bar': true}]));
 
 expectType<string[]>(camelcaseKeys(['name 1', 'name 2']));
 
 expectType<string[]>(camelcaseKeys(['name 1', 'name 2'], {deep: true}));
 
-expectType<{[key in 'foo-bar']: true}>(camelcaseKeys({'foo-bar': true}));
+expectType<{[key in 'fooBar']: boolean}>(camelcaseKeys({'foo-bar': true}));
+expectType<{[key in 'fooBar']: boolean}>(camelcaseKeys({'--foo-bar': true}));
+expectType<{[key in 'fooBar']: boolean}>(camelcaseKeys({foo_bar: true}));
+expectType<{[key in 'fooBar']: boolean}>(camelcaseKeys({'foo bar': true}));
 
-expectType<{[key in 'foo-bar']: true}>(
-	camelcaseKeys({'foo-bar': true}, {deep: true})
+expectType<{fooBar: {fooBar: {fooBar: boolean}}}>(
+	camelcaseKeys(
+		{'foo-bar': {foo_bar: {'foo bar': true}}},
+		{deep: true}
+	)
 );
 
-expectType<{[key in 'foo-bar']: true}>(
-	camelcaseKeys({'foo-bar': true}, {deep: true, pascalCase: true})
+expectType<{[key in 'FooBar']: boolean}>(
+	camelcaseKeys({'foo-bar': true}, {pascalCase: true})
+);
+expectType<{[key in 'FooBar']: boolean}>(
+	camelcaseKeys({'--foo-bar': true}, {pascalCase: true})
+);
+expectType<{[key in 'FooBar']: boolean}>(
+	camelcaseKeys({foo_bar: true}, {pascalCase: true})
+);
+expectType<{[key in 'FooBar']: boolean}>(
+	camelcaseKeys({'foo bar': true}, {pascalCase: true})
+);
+expectType<{FooBar: {FooBar: {FooBar: boolean}}}>(
+	camelcaseKeys(
+		{'foo-bar': {foo_bar: {'foo bar': true}}},
+		{deep: true, pascalCase: true}
+	)
 );
 
-expectType<{[key in 'foo-bar']: true}>(
-	camelcaseKeys({'foo-bar': true}, {exclude: ['foo', /bar/]})
+expectType<{[key in 'fooBar' | 'foo_bar']: boolean}>(
+	camelcaseKeys(
+		{'foo-bar': true, foo_bar: true},
+		{exclude: ['foo', 'foo_bar', /bar/] as const}
+	)
 );
 
-expectType<{[key in 'foo-bar']: true}>(
+expectType<{[key in 'fooBar']: boolean}>(
 	camelcaseKeys({'foo-bar': true}, {stopPaths: ['foo']})
 );
```

Comment:
Can you look at the tests in https://github.com/sindresorhus/camelcase-keys/pull/69/files#diff-724a6378efca7770e513d0ca30555fe214930f896214ac52213320a0b354df48R1 and see if there's anything there that could be added here too?